src/commands/ServiceCommands.ts

```diff
@@ -235,6 +276,98 @@ async function getExistingPorts(service: LocalService): Promise<number[]> {
   return localProjects.map((item) => item.port);
 }
 
+async function getServiceTypes(serviceTypes: TServiceType[]): Promise<TServiceType> {
+  const items: QuickPickItem[] = [];
+
+  serviceTypes.forEach(async (element) => {
+    items.push({
+      label: element.label,
+    });
+  });
+
+  const result: QuickPickItem = await showQuickPick(items, {
+    ignoreFocusOut: true,
+    placeHolder: `${Constants.placeholders.selectType}.`,
+  });
+
+  return serviceTypes.find((item) => item.label.Equals(result.label))!;
+}
+
+async function getNetworks(networks: TNetwork[]): Promise<TNetwork> {
+  const items: QuickPickItem[] = [];
+
+  networks.forEach(async (element) => {
+    items.push({
+      label: element.label,
+    });
+  });
+
+  const result: QuickPickItem = await showQuickPick(items, {
+    ignoreFocusOut: true,
+    placeHolder: `${Constants.placeholders.selectNetwork}.`,
+  });
+
+  return networks.find((item) => item.label.Equals(result.label))!;
+}
+
+async function getBlockNumber(): Promise<number> {
+  const blockNumber: string = await showInputBox({
+    ignoreFocusOut: true,
+    prompt: Constants.paletteLabels.enterBlockNumber,
+    placeHolder: Constants.placeholders.enterBlockNumber,
+    validateInput: async (value: string) => {
+      if (value.length.Equals(0)) return null;
+
+      if (!value.match(Constants.validationRegexps.onlyNumber)) return Constants.validationMessages.valueShouldBeNumber;
+
+      return null;
+    },
+  });
+
+  return Number(blockNumber);
+}
+
+async function getHostAddress(): Promise<string> {
+  const url: string = await showInputBox({
+    ignoreFocusOut: true,
+    prompt: Constants.paletteLabels.enterNetworkUrl,
+    placeHolder: Constants.placeholders.enterNetworkUrl,
+    validateInput: async (value: string) => {
+      if (value.length.Equals(0)) return Constants.validationMessages.invalidHostAddress;
+      if (!value.match(Constants.validationRegexps.isUrl)) return Constants.validationMessages.invalidHostAddress;
+
+      return null;
+    },
+  });
+
+  return url;
+}
+
+async function loadServiceType(): Promise<TServiceType[]> {
+  const networks: TServiceType[] = [
+    {
+      label: Constants.treeItemData.service.local.type.default.label,
+      isForked: Constants.treeItemData.service.local.type.default.isForked,
+      description: Constants.treeItemData.service.local.type.default.description,
+      networks: [],
+    },
+    {
+      label: Constants.treeItemData.service.local.type.forked.label,
+      isForked: Constants.treeItemData.service.local.type.forked.isForked,
+      description: Constants.treeItemData.service.local.type.forked.description,
+      networks: [
```

Comment:
this is fine, but would be nice to somehow make this automatic when you add new networks to your list? for the future. :-)

Reply:
Yeah, you right. It's not so good that way. Need to think about the better way to do that.
Thank you! :)